api4/team.go

```diff
@@ -231,6 +232,55 @@ func patchTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(patchedTeam.ToJson()))
 }
 
+func updateTeamPrivacy(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+
+	c.RequirePrivacy()
+	if c.Err != nil {
+		return
+	}
+
+	auditRec := c.MakeAuditRecord("updateTeamPrivacy", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+	auditRec.AddMeta("privacy", c.Params.Privacy)
+
+	if !c.App.SessionHasPermissionToTeam(*c.App.Session(), c.Params.TeamId, model.PERMISSION_MANAGE_TEAM) {
+		auditRec.AddMeta("team_id", c.Params.TeamId)
+		c.SetPermissionError(model.PERMISSION_MANAGE_TEAM)
+		return
+	}
+
+	if team, err := c.App.GetTeam(c.Params.TeamId); err == nil {
+		auditRec.AddMeta("team", team)
+	}
+
+	var teamType string
+	var openInvite bool
+	switch c.Params.Privacy {
```

Comment:
In [channel privacy API](https://api.mattermost.com/#tag/channels/paths/~1channels~1{channel_id}~1privacy/put), it's on the request body.  Would it make sense to do the same here?

Reply:
Thanks for that @saturninoabril .  I didn't realize there was a corresponding UpdateChannelPrivacy API.  I've modified this PR to mirror that API.  https://github.com/mattermost/mattermost-server/pull/14287/commits/045457a23d4112cf390b6ae5f244a0ae194f1ec8